data_structures/queue_using_two_stacks.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * @author [shoniavika](https://github.com/shoniavika)
+ * @file
+ *
+ * Implementation of a Queue using two Stacks.
+ */
+
+#include <cassert>
+#include <iostream>
+
+#include "./stack.h"
+
+/**
+ * Queue data structure. Stores elements in FIFO
+ * (first-in-first-out) manner.
+ *     methods:
+ *         push(int x)  :  Pushes x to the back of queue.
+ *         pop()        :  Removes an element from the front
+ *         peek()       :  Returns first element, without removing
+ *         empty()      :  Returns whether the queue is empty
+ */
+class MyQueue {
```

Comment:
```suggestion
template<typename T>
class MyQueue {
```
This will make the class generic to store any type of data